internal/services/containers/container_registry_token_password_resource.go

```diff
@@ -0,0 +1,413 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/containerregistry/mgmt/2021-08-01-preview/containerregistry"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/client"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ContainerRegistryTokenPasswordResource struct{}
+
+var _ sdk.ResourceWithUpdate = ContainerRegistryTokenPasswordResource{}
+
+type ContainerRegistryTokenPasswordModel struct {
+	TokenId   string                           `tfschema:"container_registry_token_id"`
+	Passwords []ContainerRegistryTokenPassword `tfschema:"password"`
+}
+
+type ContainerRegistryTokenPassword struct {
+	Name   string `tfschema:"name"`
+	Expiry string `tfschema:"expiry"`
+	Value  string `tfschema:"value"`
+}
+
+func (r ContainerRegistryTokenPasswordResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"container_registry_token_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTokenID,
+		},
+		"password": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 2,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							// TODO: Use below SDK enum once the following issue is resolved: https://github.com/Azure/azure-rest-api-specs/issues/18339
+							// string(containerregistry.PasswordNamePassword),
+							"password1",
+							string(containerregistry.PasswordNamePassword2),
+						}, false),
+					},
```

Comment:
Instead of a password list, that each take a name, can we instead make this two blocks: password1, password2 where the block name IS the name we pass in the the API? that way resource.password1.value is the value for the password with the name "password1"

does that makes sense? i think it would make for a better UX, WDYT?

Reply:
In this case the, the HCL would be:

```hcl
password1 {}
```

or

```hcl
password1 {
  expiry = "..."
}
```

The first form looks weired, tbh. On the other hand, to reference the value of the value of password1, users would be using `password1.0.value`.

I think the current config is better in either configuring or referencing facts.

Reply:
@katbyte That makes sense, I've made the change accordingly, please take another look:

```shell
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/containers -run='TestAccContainerRegistryTokenPassword_'
=== RUN   TestAccContainerRegistryTokenPassword_basic
=== PAUSE TestAccContainerRegistryTokenPassword_basic
=== RUN   TestAccContainerRegistryTokenPassword_complete
=== PAUSE TestAccContainerRegistryTokenPassword_complete
=== RUN   TestAccContainerRegistryTokenPassword_update
=== PAUSE TestAccContainerRegistryTokenPassword_update
=== RUN   TestAccContainerRegistryTokenPassword_requiresImport
=== PAUSE TestAccContainerRegistryTokenPassword_requiresImport
=== CONT  TestAccContainerRegistryTokenPassword_basic
=== CONT  TestAccContainerRegistryTokenPassword_update
=== CONT  TestAccContainerRegistryTokenPassword_requiresImport
=== CONT  TestAccContainerRegistryTokenPassword_complete
--- PASS: TestAccContainerRegistryTokenPassword_requiresImport (152.96s)
--- PASS: TestAccContainerRegistryTokenPassword_basic (194.71s)
--- PASS: TestAccContainerRegistryTokenPassword_complete (204.42s)
--- PASS: TestAccContainerRegistryTokenPassword_update (213.02s)
PASS
```